### PR TITLE
Deprecate the --three flag

### PR DESCRIPTION
**Pull request by oz123**
pipenv now uses python3 by default.

Following our discussion in #5304

### CODE REVIEW

File: pipenv/cli/options.py [hunk 1]
```diff
@@ -288,6 +289,11 @@ def callback(ctx, param, value):
 
 def three_option(f):
     def callback(ctx, param, value):
+        secho(
```
Comment (matteius) [inline pipenv/cli/options.py:292]: You should move this down inside the `if value is not None:` and then your tests will pass.  Right now it will always alert users, even if not passing this flag.

Comment (oz123) [reply pipenv/cli/options.py]: Done.

**Comment by oz123** [comment]
@matteius the breaks on Windows are unrelated to the change. It's that requests test again.

**Comment by matteius** [comment]
> @matteius the breaks on Windows are unrelated to the change. It's that requests test again.

The pypiserver PR will make those a non-issue, because they change to tests uvicorn and uvicorn[standard] instead of requests until we can sort out why requirementslib infers the specifier from setup metadata when none is specified.  It was clearly an edge case of that expected behavior and cross test contamination.

**Review by matteius — APPROVED** [review]
